Rover/sailboat.cpp

```diff
@@ -552,3 +575,113 @@ bool Sailboat::motor_assist_low_wind() const
+
+        // being blown backwards by the wind
+        bool  going_backwards = false;
+        {
+            float speed;
+            // only useful if we have a valid speed (e.g fails if no gps fix)
+            if (rover.g2.attitude_control.get_forward_speed(speed)) {
+                going_backwards = is_negative(speed);
+            }
+        }
+
+        // only pure sailboats get into irons
+        const bool cannot_motorsail = this->motor_state == UseMotor::USE_MOTOR_NEVER;
+
+        // only in irons if can't get power from sails
+        const bool headed = abs(wind_dir_apparent) < sail_no_go;
+
+        if (cannot_motorsail && going_backwards && headed) {
+            // in_irons_limbo_start_ms == 0  means not yet "in irons limbo"
+            if (_in_irons.start_ms == 0) {
+                // start "in irons limbo"
+                _in_irons.start_ms = millis();
+            } else {
+                // signs of signal to the rudder to rotate required direction
+                constexpr int rotate_clockwise = -1;
+                constexpr int rotate_anticlockwise = 1;
+
+                // we have been in the "in irons limbo" period for a while
+                if ((millis() - _in_irons.start_ms) >= IN_IRONS_LIMBO_MS) {
+
+                    // OK, long enough
+
+                    // fix the desired heading while in irons
+                    _in_irons.heading_rad =
+                        (currently_tacking)
+                        ? tack_heading_rad
+                        : radians(rover.g2.wp_nav.oa_wp_bearing_cd() * 0.01)
+                        ;
+
+                    // calculate which way to set the rudders to rotate the boat
+                    // towards desired heading as it is being blown backwards
+                    const int in_irons_rotate_sign =
+                        (wrap_PI(_in_irons.heading_rad - rover.ahrs.get_yaw()) >= 0.0)
+                        ? rotate_clockwise
+                        : rotate_anticlockwise
+                        ;
+
+                    // we want the boat to rotate so relax the sails
+                    // to prevent them acting as a weather vane
+                    relax_sails();
+
+                    // set the rudder for in irons steering output
+                    // gain is the amount of full rudder to apply in irons between 0 and 1
+                    // too much gain can stall the rudder and delay getting out of irons
+                    constexpr float in_irons_rudder_gain = 2.0/3.0;
+
+                    _in_irons.rudder = in_irons_rotate_sign * in_irons_rudder_gain;
+                    _in_irons.state = true;
+                    gcs().send_text(MAV_SEVERITY_INFO, "Sailboat: In Irons rotate onto %s",
+                                    (in_irons_rotate_sign == rotate_clockwise)? "port":"stbd");
+                }
+            }
+        } else { // not going backwards (anymore), so reset limbo timer.
+            _in_irons.start_ms = 0;
+        }
```

Comment:
This is such a little case is quite hard to work out which if it belongs to. It would be neater to invert the if and put this at the top. 
I think it would be this:
```
        if (!(cannot_motorsail && going_backwards && headed)) {
            // not going backwards (anymore), so reset limbo timer.
            _in_irons.start_ms = 0;

        } else {

Reply:
Whatever...